swig/csharp/CMakeLists.txt

```diff
@@ -16,15 +16,26 @@ set(GDAL_CSHARP_INSTALL_DIR
     CACHE PATH "Installation sub-directory for CSharp bindings")
 
 set(CSHARP_LIBRARY_VERSION
-    "net5.0"
+    "net6.0"
     CACHE STRING ".NET version to be used for libraries")
 set(CSHARP_APPLICATION_VERSION
-    "net5.0"
+    "net6.0"
     CACHE STRING ".NET version to be used for the sample Applications")
 set(CSHARP_CONFIG
     "RELEASE"
     CACHE STRING "Config to be used to compile the C# artifacts > RELEASE|CONFIG")
 
+string(REGEX MATCH "[0-9]+\.*[0-9]+" CSHARP_SDK ${CSHARP_LIBRARY_VERSION} )
+message(STATUS "Requested DotNet SDK Level: " ${CSHARP_SDK} )
+
+if ( DOTNET_SDKS MATCHES ".*${CSHARP_SDK}.*" ) 
```

Comment:
The leading and trailing `.*`  are pointless. And you must be sure that `${CSHARP_SDK}` contains no special characters (but even `.` is special). For an exact match, there is only `string(FIND ...)` for strings (or `IN_LISTS` for lists).

Reply:
Your comment made me realise that while it was working, it was not actually checking for incorrect SDK specifications and led to rewrite it in a more robust way